evm/src/cpu/control_flow.rs

```diff
@@ -66,20 +61,30 @@ pub fn eval_packed_generic<P: PackedField>(
     nv: &CpuColumnsView<P>,
     yield_constr: &mut ConstraintConsumer<P>,
 ) {
+    // The null special instruction must be boolean.
+    yield_constr.constraint(lv.null * (lv.null - P::ONES));
+    // Once we reach a null row, there must be only null rows.
+    yield_constr.constraint_transition(lv.null * (nv.null - P::ONES));
+
     let is_cpu_cycle: P = COL_MAP.op.iter().map(|&col_i| lv[col_i]).sum();
     let is_cpu_cycle_next: P = COL_MAP.op.iter().map(|&col_i| nv[col_i]).sum();
-    // Once we start executing instructions, then we continue until the end of the table.
-    yield_constr.constraint_transition(is_cpu_cycle * (is_cpu_cycle_next - P::ONES));
+    // Once we start executing instructions, then we continue until the end of the table
+    // or we reach dummy padding rows.
+    yield_constr
+        .constraint_transition(is_cpu_cycle * (is_cpu_cycle_next - P::ONES) * (nv.null - P::ONES));
```

Comment:
Nit: I would write it as 
```suggestion
    yield_constr
        .constraint_transition(is_cpu_cycle * (is_cpu_cycle_next + nv.null - P::ONES));
```